web/src/components/apps/Dashboard.jsx

```diff
@@ -164,33 +180,47 @@ class Dashboard extends Component {
       // no-op
     }
 
-    this.setState({
-      airgapUploader: new AirgapUploader(
-        true,
-        app.slug,
-        this.onDropBundle,
-        simultaneousUploads
-      ),
-    });
+    setAirgapUploader(
+      new AirgapUploader(true, app.slug, onDropBundle, simultaneousUploads)
+    );
   };
 
-  getAppDashboard = () => {
+  useEffect(() => {
+    if (app?.isAirgap && !airgapUploader) {
+      getAirgapConfig();
+    }
+
+    updateChecker.start(updateStatus, 1000);
+    getAppDashboardJob.start(getAppDashboard, 2000);
+    if (app) {
+      setWatchState(app);
+      getAppLicense(app);
+    }
+    return () => {
+      updateChecker.stop();
+      getAppDashboardJob.stop();
+      fetchAppDownstreamJob.stop();
+    };
+  }, []);
```

Comment:
this replaces `componentDidMount` and `componentWillUnmount`.  Original [here](https://github.com/replicatedhq/kots/blob/0276eb1d5cf21073dbc77225bc319fb93ae6fdd4/web/src/components/apps/Dashboard.jsx#L126-L145). 

```
  componentDidMount() {
    const { app } = this.props;

    if (app?.isAirgap && !this.state.airgapUploader) {
      this.getAirgapConfig();
    }

    this.state.updateChecker.start(this.updateStatus, 1000);
    this.state.getAppDashboardJob.start(this.getAppDashboard, 2000);
    if (app) {
      this.setWatchState(app);
      this.getAppLicense(app);
    }
  }

  componentWillUnmount() {
    this.state.updateChecker.stop();
    this.state.getAppDashboardJob.stop();
    this.state.fetchAppDownstreamJob.stop();
  }
```